### PR TITLE
[clcrunner] add pod ip to request header

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Adds the CLC runner pod IP to the request header to avoid any confusion with the advanced dispatching logic and pod/node IPs.
 
### Motivation

Related to https://github.com/DataDog/datadog-agent/pull/4386

### CODE REVIEW

File: pkg/util/clusteragent/clusteragent.go [hunk 1]
```diff
@@ -106,6 +109,8 @@ func (c *DCAClient) init() error {
 
 	c.clusterAgentAPIRequestHeaders = http.Header{}
 	c.clusterAgentAPIRequestHeaders.Set(authorizationHeaderKey, fmt.Sprintf("Bearer %s", authToken))
+	podIP := getLocalIP()
```
Comment (ahmed-mez) [inline pkg/util/clusteragent/clusteragent.go]: we already get the clc runner ip via the downward api https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L474

```suggestion
	podIP := config.Datadog.GetString("clc_runner_host")
```

File: pkg/util/clusteragent/clusteragent.go [hunk 2]
```diff
@@ -370,3 +375,19 @@ func (c *DCAClient) GetKubernetesMetadataNames(nodeName, ns, podName string) ([]
 
 	return metadataNames, nil
 }
+
+func getLocalIP() string {
```
Comment (ahmed-mez) [inline pkg/util/clusteragent/clusteragent.go]: ✂️

File: cmd/cluster-agent/api/v1/clusterchecks.go [hunk 3]
```diff
@@ -19,12 +19,10 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks"
 	cctypes "github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
 	"github.com/DataDog/datadog-agent/pkg/config"
+	dcautil "github.com/DataDog/datadog-agent/pkg/util/clusteragent"
```
Comment (ahmed-mez) [inline cmd/cluster-agent/api/v1/clusterchecks.go]: is the `dcautil` alias mandatory here ?

Comment (celenechang) [reply cmd/cluster-agent/api/v1/clusterchecks.go]: i think to distinguish from `"github.com/DataDog/datadog-agent/pkg/clusteragent"` ?

Comment (ahmed-mez) [reply cmd/cluster-agent/api/v1/clusterchecks.go]: indeed 👍

**Comment by ahmed-mez** [comment]
Tested it and it works as expected with DCA 1.4.0 🎉 
In the DCA client unit tests we [mock](https://github.com/DataDog/datadog-agent/blob/master/pkg/util/clusteragent/clusteragent_test.go#L235) the token config and [verify](https://github.com/DataDog/datadog-agent/blob/master/pkg/util/clusteragent/clusteragent_test.go#L101-L111) if it's in the headers, could you add a similar logic to make sure the clc runner ip is there as well?

**Comment by celenechang** [comment]
definitely, thanks for testing it out @ahmed-mez !

File: pkg/util/clusteragent/clusteragent_test.go [hunk 4]
```diff
@@ -110,6 +110,13 @@ func (d *dummyClusterAgent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	clcRunnerIP := r.Header.Get(RealIPHeader)
+	if clcRunnerIP == "" {
```
Comment (ahmed-mez) [inline pkg/util/clusteragent/clusteragent_test.go]: nit: can we assert that we got the correct value as well?

File: pkg/util/clusteragent/clusteragent_test.go [hunk 5]
```diff
@@ -110,6 +110,13 @@ func (d *dummyClusterAgent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	podIP := r.Header.Get(RealIPHeader)
+	if podIP != clcRunnerIP {
+		log.Errorf("clc runner IP not provided")
```
Comment (ahmed-mez) [inline pkg/util/clusteragent/clusteragent_test.go]:
```suggestion
		log.Errorf("wrong clc runner IP %s", podIP)
```

**Review by ahmed-mez — APPROVED** [review]

**Pull request merged**